server/rooms.js

```diff
@@ -591,37 +591,62 @@ class GlobalRoom extends BasicRoom {
 		}
 		return this.formatList;
 	}
-	get configRankList() {
-		if (Config.nocustomgrouplist) return '';
-
-		// putting the resultant object in Config would enable this to be run again should config.js be reloaded.
-		if (Config.rankList) {
-			return Config.rankList;
+	/**
+	 * @param {AnyObject} group
+	 */
+	getGroupWithType(group) {
+		let {symbol, id, name} = group;
+		let type;
+		if (group.editroom || group.makeroom || group.root) {
+			type = 'leadership';
+		} else if (!group.declare && (group.mute || group.ban)) {
+			type = 'staff';
+		} else if (group.punishgroup) {
+			type = 'punishment';
+		} else if (!group.broadcast) {
+			type = 'default';
+		} else {
+			type = 'normal';
 		}
-		let rankList = [];
-
-		for (let rank in Config.groups) {
-			if (!Config.groups[rank] || !rank) continue;
+		return {symbol, id, name, type};
+	}
+	get configGroupList() {
+		// Putting the resultant object in Config would enable this to be run again should config.js be reloaded.
+		if (Config.rankList) return Config.rankList;
 
-			let tarGroup = Config.groups[rank];
-			let groupType = tarGroup.addhtml || (!tarGroup.mute && !tarGroup.root) ? 'normal' : (tarGroup.root || tarGroup.declare) ? 'leadership' : 'staff';
+		const typeOrder = ['punishment', 'default', 'normal', 'staff', 'leadership'];
+		let rankList;
 
-			rankList.push({symbol: rank, name: (Config.groups[rank].name || null), type: groupType}); // send the first character in the rank, incase they put a string several characters long
-		}
-
-		const typeOrder = ['punishment', 'normal', 'staff', 'leadership'];
+		if (Config.grouplist) {
+			// a and b are implicitly any
```

Comment:
And this is why I ask for people to add comments for these! :) The 'implicitly any' error doesn't mean you should be ignoring, it means you need to specify the types explicitly (it can't infer them so its implicitly giving them the `any` type)

Reply:
This is where the type is just:

```ts
interface TypedGroupInfo {
  type: 'punishment' | 'default' | 'normal' | 'staff' | 'leadership';
  symbol: '~' | '#' | '&' | '@' | '%' | '*' | '+' | ...; // or just string
  id: ID;
  name: string;
}